Updates to nasa_exoplanet_archive module to make it compatible with Archive 2.0

rickynilsson opened the pull request:
The module has been updated to query the new PS and PSCOMPPARS tables using TAP service, and support for querying the old tables (EXOPLANETS, EXOMULTPARS, and COMPOSITEPARS) has been dropped (with error messages pointing the user to the new tables and documentation of recent changes to the tables and their accessibility). Tests have been updated and verified, and the module documentation has also been updated.

pep8speaks commented:
Hello @rickynilsson! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`astroquery/nasa_exoplanet_archive/core.py`](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py):

> [Line 30:1](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L30): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 50:26](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L50): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 98:1](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L98): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 104:55](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L104): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 105:39](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L105): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 106:9](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L106): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 113:54](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L113): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 114:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L114): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 115:182](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L115): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 118:1](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L118): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 180:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L180): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 182:1](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L182): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 185:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L185): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 220:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L220): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 222:72](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L222): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 225:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L225): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 227:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L227): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 322:88](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L322): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 552:109](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L552): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 566:1](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L566): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 573:13](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L573): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 624:5](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/core.py#L624): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)

* In the file [`astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py`](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py):

> [Line 26:62](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L26): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 26:98](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L26): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 98:52](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L98): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 100:48](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L100): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 102:53](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L102): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 113:18](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L113): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 117:37](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L117): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 118:42](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L118): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 123:14](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L123): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 127:31](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L127): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 129:62](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L129): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 132:24](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L132): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 134:46](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L134): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 157:69](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L157): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 157:71](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L157): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 212:19](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L212): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 230:21](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L230): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 242:25](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L242): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 253:19](https://github.com/astropy/astroquery/blob/abd255ea3868f012b0863a41a57f180cdee75a54/astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py#L253): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment

bsipocz commented:
@rickynilsson - I don't see an obvious reason for the cancelled jobs (they will be cancelled when there one of the fails, but haven't seen any of them got to that stage yet), I've restarted the workflow.

rickynilsson commented:
> @rickynilsson - I don't see an obvious reason for the cancelled jobs (they will be cancelled when there one of the fails, but haven't seen any of them got to that stage yet), I've restarted the workflow.

Now codestyle check failed. Just fixed those issues and committed again.

rickynilsson commented:
@bsipocz - the oldest dependencies (Python 3.6) tests should pass, as far as I can see. They run fine in my Python 3.6 environment, and https://exoplanetarchive.ipac.caltech.edu/cgi-bin/nstedAPI/nph-nstedAPI?table=aliastable&objname=HAT-P-11+b&format=csv responds as expected (as you can verify). Not sure how to troubleshoot this one.

bsipocz commented:
The issue with that test (that will come up in all the other runs, too) that they are not marked with `remote_data` yet they reach out to the server. By default, we don't let any internet access happen. 
I suppose if you run the tests locally you would see the same failure.

To fix it, decorate the test functions that reach out to the server with `@pytest.mark.remote_data`, see examples in `modulename/tests/test_modulename_remote.py` files.

rickynilsson commented:
@bsipocz - I decorated the remote tests and moved them over to the `test_..._remote.py` file where they belongs. I also added some more tests for increased code coverage. Verified test on my side: 43 passed, 19 warnings (these are mainly `PytestUnknownMarkWarning: Unknown pytest.mark.remote_data` for some reason). Codecov should hopefully be happier now.

rickynilsson commented:
Hi @bsipocz, @keflavich, @ceb8,
Just tagging you to see if one of you can approve the workflow for the merge. This is currently one of the NASA Exoplanet Archive's high priority items, and we are hoping to announce the updated module to the exoplanet community soon (hopefully even in today's newsletter). Thanks!

rickynilsson commented:
@bsipocz, @keflavich, @ceb8 - It appears checks were automatically cancelled for no obvious reasons (no failed check). This happened once before, but then went through when workflow was restarted.

rickynilsson commented:
@bsipocz, @keflavich, @ceb8: CI / Python 3.8 with all dependencies with remote data (pull_request) got skipped and rest got cancelled again. Any idea what could be going on?

bsipocz commented:
I'm not sure what's going on with these actions, I've just restarted the workflow but it got cancelled again.

rickynilsson commented:
> I'm not sure what's going on with these actions, I've just restarted the workflow but it got cancelled again.

Weird...
@bsipocz - Any chance I could get write access to re-run the actions myself, so that I don't have to keep bugging you? Or is there any workaround for merging this without passed checks?

rickynilsson commented:
Thanks! I'll have to run Codecov locally and try to get the coverage up.

rickynilsson commented:
@bsipocz - Coverage.py reports a total test coverage of 96%
```
---------- coverage: platform darwin, python 3.8.8-final-0 -----------
Name                                                                 Stmts   Miss  Cover
----------------------------------------------------------------------------------------
nasa_exoplanet_archive/__init__.py                                      10      0   100%
nasa_exoplanet_archive/core.py                                         225     14    94%
nasa_exoplanet_archive/tests/__init__.py                                 1      0   100%
nasa_exoplanet_archive/tests/setup_package.py                            3      3     0%
nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py             56      0   100%
nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive_remote.py     105      0   100%
----------------------------------------------------------------------------------------
TOTAL                                                                  400     17    96%
```
Not sure how Codecov calculates its numbers.

rickynilsson commented:
@bsipocz -  I wonder if the unexpectedly low diff coverage value comes from removed tests (most pertaining to the now unsupported tables, and deprecated methods, and some mock-data tests moved to remote tests). See https://docs.codecov.io/docs/unexpected-coverage-changes

I've added a few more tests to get the total coverage up to 97%. I'm happy to send you my coverage.xml generated from running:
`pytest --cov=nasa_exoplanet_archive nasa_exoplanet_archive/tests/`
`coverage xml`

Hoping to get over this Codecov hurdle somehow.

bsipocz commented:
> @bsipocz - Any chance I could get write access to re-run the actions myself, so that I don't have to keep bugging you? Or is there any workaround for merging this without passed checks?

Only the maintainers' team has write access to repos, not individuals. But anyways, write access is not needed, the CI is not run automatically due to github's being attached by crypto miners, and they turned off CI starting up for first-time contributors. 

The known workarounds are: 
- make a small PR that we can merge asap with a minor change, anywhere. 
- or, add you to the large astropy contributors team. 

I would love to do the latter (already tried it), but it's only the coordination committee is gatekeeping the access rights from the rest of the team to be able to add new people. So we need to fall back to the PR workaround. Any small dummy PR with a single commit would do, e.g. fix the alphabetical order for nasa_exoplanet_archive in docs/index.rst.

As for coverage: it sometimes behaves weirdly with bogus looking drop in coverage when the base of the PR and the latest run on master differs. I think simply there was a lot of cleanups in the recent 0.4.2. Also, you have restructured the tests, so it's possible that some parts of the code now covered only with `remote-data` tests, and the coverage run doesn't pick those up.
In summary: we look into the coverage status during review, but merge a PR even with a failing status if everything otherwise looks good (and the status can be easily explained with one of the scenarios above).

rickynilsson commented:
> > @bsipocz - Any chance I could get write access to re-run the actions myself, so that I don't have to keep bugging you? Or is there any workaround for merging this without passed checks?
> 
> Only the maintainers' team has write access to repos, not individuals. But anyways, write access is not needed, the CI is not run automatically due to github's being attached by crypto miners, and they turned off CI starting up for first-time contributors.
> 
> The known workarounds are:
> 
> * make a small PR that we can merge asap with a minor change, anywhere.
> * or, add you to the large astropy contributors team.
> 
> I would love to do the latter (already tried it), but it's only the coordination committee is gatekeeping the access rights from the rest of the team to be able to add new people. So we need to fall back to the PR workaround. Any small dummy PR with a single commit would do, e.g. fix the alphabetical order for nasa_exoplanet_archive in docs/index.rst.
> 
> As for coverage: it sometimes behaves weirdly with bogus looking drop in coverage when the base of the PR and the latest run on master differs. I think simply there was a lot of cleanups in the recent 0.4.2. Also, you have restructured the tests, so it's possible that some parts of the code now covered only with `remote-data` tests, and the coverage run doesn't pick those up.
> In summary: we look into the coverage status during review, but merge a PR even with a failing status if everything otherwise looks good (and the status can be easily explained with one of the scenarios above).

Thanks @bsipocz! I made the suggested edit in docs/index.rst and submitted a PR, so hopefully I can run CI workflows soon. That will also be helpful over the next couple of months as we move more of our tables over to the TAP service.

rickynilsson commented:
@bsipocz - Any idea when merge review might take place? Sorry for being impatient :-)

rickynilsson commented:
@bsipocz, @keflavich, @ceb8, @dfm, @bmorris3 -
Hi ya'll. Just wanted to check in again about when merge review for these Archive 2.0 updates might happen. Cheers!

bsipocz commented:
@rickynilsson - I'm sorry, I'm horribly swamped at the moment with the end of quarter and wrapping up my current day job. It may take a few weeks on my side to get enough time for a proper, thorough PR review.

dfm commented:
I don't know enough about the astroquery requirements to do a full review, but I'm happy to take a stab at a review from a user perspective if that would be helpful, @bsipocz!

dfm commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -115,11 +142,24 @@ def query_criteria_async(self, table, get_query_payload=False, cache=None, **cri
+            raise InvalidTableError("The ``{0}`` table is no longer updated and has been replaced by the Planetary Systems table (PS), which is connected to the Exoplanet Archive TAP service. Although the argument keywords of the called method should still work on the new table, the allowed values could have changed since the database column names have changed; this document contains the current  ...
+
+        # Warn if old table is requested
+        if table in ["compositepars"]:
```
It might be worth combining this with the above condition and just mapping the updated table names using a dict or something?

dfm commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -246,9 +304,14 @@ def query_object_async(self, object_name, *, table="exoplanets", get_query_paylo
+        .. [2] `NASA Exoplanet Archive API Documentation
            <https://exoplanetarchive.ipac.caltech.edu/docs/program_interfaces.html>`_
         """
+        # if table.lower() in ["ps"]: # actually want to check if default was used, but wasn't working ...
```
It would be possible to use `table=None` and then show this warning, but this could also be annoying as a user. Someone who knows more about the astroquery ecosystem should comment on how best to handle deprecations like this.

keflavich replied:
I'm pretty sure we have deprecation warning decorators.  But I'm not sure how important a warning this is.  Should users expect dramatically different behavior?

rickynilsson replied:
Querying the PS table will generally return multiple rows per object, as opposed to only one per object for the old EXOPLANETS table, which is why we wanted the warning there if the new default is used. If specifically querying the old tables we want the code to break with an error message detailing the change in Archive 2.0, rather than just throwing a deprecation warning.

rickynilsson replied:
@dfm, @keflavich - I've added a few mock tests for TAP. Let me know if you've had time to test out the functionality and are ready to merge this fix. The NExScI managers are getting impatient :P

keflavich replied:
Was this comment meant for this specific line?  It looks like there is no warning here, right?

dfm commented on astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py:
```diff
@@ -7,26 +7,32 @@
 MAIN_DATA = pkg_resources.resource_filename("astroquery.nasa_exoplanet_archive", "data")
 TEST_DATA = pkg_resources.resource_filename(__name__, "data")
 RESPONSE_FILE = os.path.join(TEST_DATA, "responses.json")
+os.environ["NASA_EXOPLANET_ARCHIVE_GENERATE_RESPONSES"] = '1'  # Activate for generating responses
```
This shouldn't be set here. This environment variable should *only* be set on your local machine when *creating* the tests. This was a hack that I used to automatically generate the correct mocked responses, but shouldn't ever be used when running the tests.

dfm commented on astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive_remote.py:
```diff
@@ -53,3 +59,114 @@ def test_missing_criterion_super_wasp():
     )
+
+
+@pytest.mark.remote_data
```
This mark isn't needed.

dfm reviewed (commented):
I'll try actually using this interface a little later, but it looks to me like the big remaining issue is that there appear to be no tests for the TAP interface that don't rely on remote data. Others can comment on whether or not that is okay, but I would certainly prefer to be able to test the interface using mocked responses.

rickynilsson commented:
Thanks, @dfm ! I honestly couldn't figure out exactly what was going on in `patch_get(request)`, which is why I left out mock tests for the TAP service. I didn't know they were necessary, and I suspect Codecov checks fail because of removed tests of deprecated tables (giving a low diff coverage). If you would want to help out setting up the mock tests for TAP that would be greatly appreciated. I'll clean up the other issues you mentioned.

dfm commented:
@rickynilsson: Sure - it's a little opaque, I know! I'll send some demo tests to give you a sense of it.

keflavich commented:
@andamian provides some good examples of mocking tests for TAP in https://github.com/astropy/astroquery/blob/main/astroquery/cadc/tests/test_cadctap.py

dfm commented:
@rickynilsson: Looking through this more, I honestly think that a better interface design might be to have separate `NasaExoplanetArchive` and `NasaExoplanetArchiveTAP` classes. I worry that the current implementation is confusing in how it tries to abstract such different interfaces into one class. It might be better for users to be more explicit. Then the `NasaExoplanetArchive` interface can just correctly throw exceptions for the queries that are no longer supported instead of trying to shoehorn into the new interface. This would also make the testing (using the reference shared by @keflavich) and implementation simpler. What do you think about that?

rickynilsson commented:
@bsipocz - totally understand that you are busy with work! Maybe if @dfm helps out with a quick review from a user perspective, we can merge fairly soon, and fix any bugs or lack of astroquery requirements (should be minor) later? We haven't touched any other modules, and thoroughly tested the restored functionality of our module. Note that the nasa_exoplanet_archive module in main is currently broken, and we are starting to get questions from users about it. Alternatively, we could direct users to our working branch on https://github.com/IPAC-SW/astroquery/tree/nasa_exoplanet_archive_2 , if that is okay with you.

keflavich commented:
@dfm ideally, we don't want the users to have to know anything about TAP unless it's strictly necessary.  Splitting into multiple classes would require users to make that choice, no?

rickynilsson commented:
@dfm , @keflavich - The Archive is gradually moving all tables over to the TAP service, so splitting into two classes might be unnecessary.

keflavich commented:
with Dan's review & approval, and given that the existing tool is broken, I'll be happy to merge this sooner than @bsipocz can.

keflavich commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -262,15 +321,19 @@ def query_object_async(self, object_name, *, table="exoplanets", get_query_paylo
             table, get_query_payload=get_query_payload, cache=cache, **criteria,
         )
 
+    # This should stay the same for now
```
the same as what?

rickynilsson replied:
That was a leftover comment from my initial review. Removing.

keflavich commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -420,18 +496,13 @@ def _fix_units(self, data):
 
-            # Here we're figuring out out if the column is masked because this doesn't
-            # play nice with quantities so we need to keep track of the mask separately.
+            # Unmask since astropy doesn't like masked values in columns with units
```
😱 really?!  that's annoying.  (you can mark this resolved, I just had to remark on it)

keflavich commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -510,6 +586,30 @@ def _handle_all_columns_argument(self, **kwargs):
+            del request_payload["radius"]
+        if "where" in request_payload:
+            if "pl_hostname" in request_payload["where"]:  # means this is a `query_object`
+                request_payload["where"] = "pl_hostname or pl_name like {0}".format(request_payload["where"][request_payload["where"].find("=")+2:request_payload["where"].find("OR")-2])  # This is a bit hacky since we are getting this from the request_payload (downstream) instead of directly from object_name
```
could you make this fit onto one line?  and line 600 above

rickynilsson replied:
Fixed.

keflavich commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -510,6 +586,30 @@ def _handle_all_columns_argument(self, **kwargs):
+
+        return tap_query
+
+    # Could we use similar @deprecated decorator for new changes and warnings?
```
yes, I think so?

keflavich replied:
(this might be the answer to @dfm's question above about how to deprecate)

rickynilsson replied:
I should remove that comment. We (the Archive) decided that an InvalidTableError would be better, since the old tables should be considered removed rather than deprecated.

bsipocz commented:
> with Dan's review & approval, and given that the existing tool is broken, I'll be happy to merge this sooner than @bsipocz can.

a rebase would be nice to cleanup the merge commits, but otherwise with @dfm's user tests and your quick review I would be OK with going ahead, given it's fixing the module that is currently broken.

dfm commented:
@ everyone - sounds good re: not wanting to split into two classes. I'll keep going through the new interface and try to send some demo tests later today!

ceb8 commented:
I can give this a review Friday as well.

dfm commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -143,14 +181,26 @@ def query_criteria_async(self, table, get_query_payload=False, cache=None, **cri
+        # The _request method is a custom astroquery wrapper around the requests.request function that provides important astroquery-specific utility, including caching, HTTP header generation, progressbars, and local writing-to-disk.
+        # This needs to be updated to use TAP (TapPlus), and the request_payload formatted correctly
+        # Execute the request (generic HTTP request, similar to requests.Session.request but with added caching-related tools)
+        if table in ["ps", "pscomppars"]:
```
An issue with this interface (the TAP table names being hard coded) is that as more tables are migrated new versions of astroquery will need to be released. A better way might be to hard code the API table names and have the *default* be TAP. What are the chances that TAP tables with the same names as API tables will be released?

dfm replied:
Similarly, I think that it might be worth adding a `force_tap` parameter or something to make it possible to override the default behavior unless we're confident that we can cover all the future tables!

rickynilsson replied:
That sounds sensible. I think the chances that we'll duplicate table names are pretty low (to avoid confusion within the archive too). I'll make this change.

rickynilsson replied:
Actually it seems some of the table names will remain the same. Could I try given table name against the TAP service, and if it returns an error, try against the API service?

rickynilsson replied:
I'll make use of: https://exoplanetarchive.ipac.caltech.edu/TAP/sync?query=select+schema_name,table_name+from+TAP_SCHEMA.tables&format=csv

dfm replied:
Awesome - yeah that would be a great way to do it, I think!

rickynilsson replied:
Did this:
```
# Tables accessed by API are gradually migrating to TAP service. Generate current list of tables in TAP.
tap = pyvo.dal.tap.TAPService(baseurl=conf.url_tap)
response = tap.search(query="select * from TAP_SCHEMA.tables", language="ADQL")
TAP_TABLES = [table for table in response["table_name"].data if "TAP_SCHEMA." not in table]
```

dfm commented on astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py:
```diff
@@ -68,44 +72,63 @@
-    # Work out where the expected response is saved
     table = params["table"]
-    key = urlencode(sorted(params.items()))
+    # Set base URL depending on whether TAP or API for access
```
re: tests for TAP interface. From the experiments I've tried, I think it would be worth reverting these changes to `mock_get` and then write separate test cases for the TAP API that follow the examples here: https://github.com/astropy/astroquery/blob/main/astroquery/cadc/tests/test_cadctap.py

It seems like it would be possible to update this function to do what we need, but it might be more trouble than it's worth.

rickynilsson replied:
Yeah it started getting messy, so that is probably a good idea. Thanks!

rickynilsson replied:
Done!

rickynilsson commented:
@dfm, @keflavich - I'm getting "`An attempt was made to connect to the internet by a test that was not marked `remote_data``". As far as I can see, all the tests that should need the @pytest.mark.remote_data decorator have it (and probably some that don't need it). Any ideas?

keflavich commented:
which test is triggering that?

rickynilsson commented:
> which test is triggering that?

py36-test-oldestdeps run-test: commands[1] | pytest --pyargs astroquery

rickynilsson commented:
> > which test is triggering that?
> 
> py36-test-oldestdeps run-test: commands[1] | pytest --pyargs astroquery
> 
> I can't tell exactly where it is coming from.

@keflavich - It appears that the tests didn't like the added
```
    tap = pyvo.dal.tap.TAPService(baseurl=conf.url_tap)
    response = tap.search(query="select * from TAP_SCHEMA.tables", language="ADQL")
    tables = [table for table in response["table_name"].data if "TAP_SCHEMA." not in table]
    return tables
```
in core.py, so I went back to hard coding the TAP tables. Would be neat if we could get it to work, but in either case, we can't expect to fully future proof the module at the moment. There will be some degree of periodic updates to keep it in sync with our migration to TAP, and as we've mentioned before, a possible larger overhaul down the road to get all IPAC project modules under one umbrella. The goal right now is just to get this patched module out.

keflavich reviewed (approved):
@bsipocz @ceb8 I suggest we merge this now, and @rickynilsson can come back with bugfixes in smaller PRs if needed.

bsipocz commented:
@keflavich @rickynilsson - Please hold back merging until this is rebased to get rid of the merge commits. Thanks!

dfm commented:
I personally still think it's a mistake to ship this with hard-coded TAP table names, but otherwise I think it's looking great! I'm also very much not in charge so please take what I say with a grain of salt, but I expect that requiring a new astroquery version for every update at IPAC will end up as a nightmare for users! I guess the severity of this issue depends on the future migration schedule for the archive.

It would be possible to fix the remote data issues by mocking the table name query. I think that would be a better solution than reverting to hard coding.

rickynilsson commented:
> @keflavich @rickynilsson - Please hold back merging until this is rebased to get rid of the merge commits as per [#2067 (comment)](https://github.com/astropy/astroquery/pull/2067#issuecomment-848975594). Thanks!

I'm getting stuck here:
```
git rebase --continue                                                   
Applying: MNT: Replace deprecated np.str
No changes - did you forget to use 'git add'?
If there is nothing left to stage, chances are that something else
already introduced the same changes; you might want to skip this patch.
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```
I did solve the conflict and ran `git add astroquery/nasa_exoplanet_archive/core.py`, so not sure why the complaint.

rickynilsson commented:
> I personally still think it's a mistake to ship this with hard-coded TAP table names, but otherwise I think it's looking great! I'm also very much not in charge so please take what I say with a grain of salt, but I expect that requiring a new astroquery version for every update at IPAC will end up as a nightmare for users! I guess the severity of this issue depends on the future migration schedule for the archive.
> 
> It would be possible to fix the remote data issues by mocking the table name query. I think that would be a better solution than reverting to hard coding.

@dfm - I'd be happy to add this. Any example on how that would look? Thanks

rickynilsson commented:
> > @keflavich @rickynilsson - Please hold back merging until this is rebased to get rid of the merge commits as per [#2067 (comment)](https://github.com/astropy/astroquery/pull/2067#issuecomment-848975594). Thanks!
> 
> I'm getting stuck here:
> 
> ```
> git rebase --continue                                                   
> Applying: MNT: Replace deprecated np.str
> No changes - did you forget to use 'git add'?
> If there is nothing left to stage, chances are that something else
> already introduced the same changes; you might want to skip this patch.
> Resolve all conflicts manually, mark them as resolved with
> "git add/rm <conflicted_files>", then run "git rebase --continue".
> You can instead skip this commit: run "git rebase --skip".
> To abort and get back to the state before "git rebase", run "git rebase --abort".
> ```
> 
> I did solve the conflict and ran `git add astroquery/nasa_exoplanet_archive/core.py`, so not sure why the complaint.

I proceeded with `git rebase --skip`, which seems to have done the trick.

bsipocz commented:
@rickynilsson - yes, I see a couple of trivial conflicts in `astroquery/nasa_exoplanet_archive/core.py` for multiple commits, but otherwise the rebase seems to be a smooth one. Also, you may want to revise the commit message for the first commit, it seems a `ps` got added into it (if you rebase interactively, you can choose the edit for that commit).

rickynilsson commented:
> > > which test is triggering that?
> > 
> > 
> > py36-test-oldestdeps run-test: commands[1] | pytest --pyargs astroquery
> > I can't tell exactly where it is coming from.
> 
> @keflavich - It appears that the tests didn't like the added
> 
> ```
>     tap = pyvo.dal.tap.TAPService(baseurl=conf.url_tap)
>     response = tap.search(query="select * from TAP_SCHEMA.tables", language="ADQL")
>     tables = [table for table in response["table_name"].data if "TAP_SCHEMA." not in table]
>     return tables
> ```

@keflavich, @dfm - I'm basically back at this issue now, even after adding a mock test for the table name query. Any suggestions on the last commit?

dfm commented:
I'm offline tonight, but I'll try to take a look in the morning!

rickynilsson commented:
> I'm offline tonight, but I'll try to take a look in the morning!
> 
> Please don't let me hold this up - sorry!

No worries, I totally agree with you regarding fetching the tables names instead of hard coding them -- I just can't get it to pass pytest :)

keflavich commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -59,10 +73,39 @@
+    return tables
+
+
+TAP_TABLES = get_tap_tables(conf.url_tap)  # TAP_TABLES = ['ps', 'pscomppars']
```
This line is the problem: you can't (by astroquery rules) perform a remote query at import time.

rickynilsson replied:
Thanks @keflavich, that makes perfect sense.

rickynilsson replied:
@keflavich - I think we are pretty close now. TAP_TABLES is now an attribute of the class, and get_tap_tables is also a method of the class. Everything seems to work in terms of module functionality, only issue is the testing which now insists on trying to use the TAP service for all the tests, and failing with `AssertionError: assert 'https://exop...edu/TAP//sync' == 'https://exop.../nph-nstedAPI' `. It isn't clear to me why fetching the TAP table names in core.py introduces this change in `test_regularize_object_name`, `test_backwards_compat`, and `test_api_tables` . Any idea?

keflavich replied:
Yeah, it looks like those tests have monkeypatched the `Session.request`, but they are calling functions that try to use the TAP URL and therefore call the `TAP_TABLES` attribute - so they're trying to send a request.   You probably need to monkeypatch the `TAP_TABLES` attribute in those tests.

rickynilsson replied:
Ah, I see what you mean. Alright, let me see if I can get that fixed.

rickynilsson replied:
My naive
```
    # Patch TAP_TABLES
    mp.setattr(__name__+".NasaExoplanetArchive.TAP_TABLES", property(lambda: ['notap']))
```
in `patch_get` didn't seem to work, and then I went down a very deep rabbit hole in which I'm still stuck. Is monkeypatching supposed to be this hard? I just want to set the TAP_TABLES property to an empty list in these tests of API tables. It seems like `mp.setattr` still runs the TAP query. Part of error message:

```
...
../../../../_pytest/monkeypatch.py:100: in derive_importpath
    annotated_getattr(target, attr, ann=module)
../../../../_pytest/monkeypatch.py:84: in annotated_getattr
    obj = getattr(obj, name)
../core.py:134: in TAP_TABLES
    self._tap_tables = get_tap_tables()
../core.py:103: in get_tap_tables
    response = tap.search(query="select * from TAP_SCHEMA.tables", language="ADQL")
...
```

keflavich replied:
the problem is, I think, with that `__name__` thing - why are you doing that?

You can do something much simpler here - technically you don't even need to monkeypatch for this case:
```python
def test_regularize_object_name(patch_get):
    NasaExoplanetArchive._tap_tables = ['list']
```

but if you were going to rename an attribute, it would be more like:
```python
mp.setattr(NasaExoplanetArchive, '_tap_tables', ['list'])
```

keflavich replied:
I'm wrong about the monkeypatching approach... not entirely sure what you would need to do for that, but it's something different here.  Possibly would need `lambda x: ['notap']` because it expects to receive `self` (it's a method on a class instance), but that might require going one level deeper... in any case, you don't need this here.

rickynilsson replied:
The `__name__` thing was because I was getting a Module Not Found error, I assume due to the relative import of `NasaExoplanetArchive` from `..core`. I'll try your suggestion. Thanks!

keflavich replied:
in this case, `NasaExoplanetArchive` is actually an _instance_ of a class everywhere you're using it, so you don't need to monkeypatch it.  Monkeypatching is for changing things that are harder to access.... for example, in this case, I think you actually wanted to patch `get_tap_tables`, e.g.:
```python
@patch('astroquery.nasa_exoplanet_archive.core.get_tap_tables',
       lambda x: ['notap'])
```

that seems to also work (and means you can test the `TAP_TABLES` attribute)

rickynilsson replied:
Thanks @keflavich. The checks are finally all green (except Codecov not quite on target, but hopefully you can let that slide). Now I just have to fix that crazy first commit message, and we are set.

bsipocz replied:
Don't worry about codecov, its status is mostly used informally.

keflavich reviewed (changes requested):
I'm not entirely sure I understand the problem, but it looks to me like you're running a query at import time.  That is not allowed, and you can't simply mock it away - this isn't a testing problem, this is a problem that _every time_ someone imports the module, it will automatically run a query, which is bad!

A solution is to make `TAP_TABLES` an attribute of `NasaExoplanetArchiveClass`, e.g.:

```python
class NasaExoplanetArchiveClass(BaseQuery):
    @property
    def TAP_TABLES(self):
        if not hasattr(self, '_tap_tables'):
            self._tap_tables = get_tap_tables(conf.url_tap)
        return self._tap_tables
```

unless there's some reason that `TAP_TABLES` needs to be a module-level property?

rickynilsson commented:
> @rickynilsson - yes, I see a couple of trivial conflicts in `astroquery/nasa_exoplanet_archive/core.py` for multiple commits, but otherwise the rebase seems to be a smooth one. Also, you may want to revise the commit message for the first commit, it seems a `ps` got added into it (if you rebase interactively, you can choose the edit for that commit).

@bsipocz - Ouch, that will teach me not to use back-tick in commit messages. When I clean that up using interactive rebase, I get a bunch of conflicts CHANGES.rst:

```
$ git rebase -i HEAD~106                           
Auto-merging docs/index.rst
Auto-merging README.rst
Auto-merging CHANGES.rst
CONFLICT (content): Merge conflict in CHANGES.rst
error: could not apply 5cad957e... Clean commit version of ISO support
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 5cad957e... Clean commit version of ISO support
```

Skipping just brings up the next conflict.

bsipocz commented:
I haven't seen any conflicts with the CHANGES at all, only with the `core.py` file above. But indeed, fixing it once will bring it up a couple of more times.  
Either way, I'm happy to force push the rebased branch here, once it's fully ready and gets the passing review from Adam.

rickynilsson commented:
Now the API query tests are trying to use the TAP base URL. Trying to figure out why ...

rickynilsson commented:
@bsipocz - For the interactive rebase, after I add "reword" and update the bungled commit messages, it keeps asking me to resolve conflicts, basically stepping through various commits. Should I just skip all those? The interactive rebasing is making me a bit nervous, just don't want to screw anything up this close to the finish line.

bsipocz commented:
I think the main reason for that many conflicts is that a rebase attempt didn't work yesterday and a few duplicated commits from the PR as well as a few unrelated ones from `main` got pushed back to this branch. 

(Otherwise, it's quite normal that some trivial conflicts come up with multiple commits during the same rebase process. Basically, you need to resolve those as you go along with the rebase with `git rebase --continue` until you're done. If you want an extra safety measure, make a copy branch before you start, I found that approach helpful to tune down nervousness whenever I  have a more complex case. If all goes wrong I can easily go back to it as a backup).

rickynilsson commented:
@bsipocz - After the rebase, it says my branch and origin/'origin/nasa_exoplanet_archive_2' have diverged, which I guess is expected. Do I want to do a `git pull`?

```
Successfully rebased and updated refs/heads/nasa_exoplanet_archive_2.
~/Documents/external_repos/astroquery » git status
On branch nasa_exoplanet_archive_2
Your branch and 'origin/nasa_exoplanet_archive_2' have diverged,
and have 87 and 128 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours
```

rickynilsson commented:
> @bsipocz - After the rebase, it says my branch and origin/'origin/nasa_exoplanet_archive_2' have diverged, which I guess is expected. Do I want to do a `git pull`?
> 
> ```
> Successfully rebased and updated refs/heads/nasa_exoplanet_archive_2.
> ~/Documents/external_repos/astroquery » git status
> On branch nasa_exoplanet_archive_2
> Your branch and 'origin/nasa_exoplanet_archive_2' have diverged,
> and have 87 and 128 different commits each, respectively.
>   (use "git pull" to merge the remote branch into yours
> ```

Actually, I guess I'd want to push the local branch since it has the changed history?

bsipocz commented:
The side effect of doing a `pull` in these cases is the appearance of duplicated and unrelated commits. Instead, you want a push, and it has to be a `--force` one as you purposefully rewrite the history of the remote.

rickynilsson commented:
@bsipocz - I force pushed, then resolved conflicts with main (in GitHub). Then I did a git pull on my my local branch to incorporate those changes, but something doesn't look right.

```
~/Documents/external_repos/astroquer » git pull  
remote: Enumerating objects: 575, done.
remote: Counting objects: 100% (472/472), done.
remote: Compressing objects: 100% (177/177), done.
remote: Total 405 (delta 271), reused 340 (delta 223), pack-reused 0
Receiving objects: 100% (405/405), 356.93 KiB | 1.32 MiB/s, done.
Resolving deltas: 100% (271/271), completed with 37 local objects.
From https://github.com/IPAC-SW/astroquery
   fab539cd..55710fb0  nasa_exoplanet_archive_2 -> origin/nasa_exoplanet_archive_2
Updating fab539cd..55710fb0
Fast-forward
 CHANGES.rst                                                 |  44 ++++++++-
 astroquery/esa/hubble/__init__.py                           |   4 +
 astroquery/esa/hubble/core.py                               | 330 ++++++++++++++++++++++++++++++++++++++++++++++++++--------------
 astroquery/esa/hubble/tests/data/cone_search.vot            | 324 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 astroquery/esa/hubble/tests/data/cone_search_m31_5.vot      | Bin 0 -> 288681 bytes
 astroquery/esa/hubble/tests/data/cone_search_m31_5.vot.test | Bin 0 -> 288681 bytes
 astroquery/esa/hubble/tests/dummy_handler.py                |  71 --------------
 astroquery/esa/hubble/tests/dummy_tap_handler.py            |  44 ---------
 astroquery/esa/hubble/tests/test_esa_hubble.py              | 212 +++++++++++++++++++++++++++++++++--------
 astroquery/esa/hubble/tests/test_esa_hubble_remote.py       | 101 ++++++++++++++++++++
 astroquery/heasarc/__init__.py                              |   3 +-
 astroquery/heasarc/core.py                                  | 101 ++++++++++++++------
 astroquery/heasarc/tests/test_heasarc_remote.py             |  28 ++++++
 astroquery/heasarc/tests/test_heasarc_remote_isdc.py        | 209 +++++++++++++++++++++++++++++++++++++++++
 astroquery/lcogt/__init__.py                                |  41 --------
 astroquery/lcogt/core.py                                    | 441 --------------------------------------------------------------------------------------
 astroquery/lcogt/tests/__init__.py                          |   0
 astroquery/lcogt/tests/data/Box.xml                         |  70 --------------
 astroquery/lcogt/tests/data/Cone.xml                        |  72 --------------
 astroquery/lcogt/tests/data/Cone_coord.xml                  |  72 --------------
 astroquery/lcogt/tests/data/Polygon.xml                     |  76 ---------------
 astroquery/lcogt/tests/setup_package.py                     |  10 --
 astroquery/lcogt/tests/test_lcogt.py                        | 177 -----------------------------------
 astroquery/lcogt/tests/test_lcogt_remote.py                 |  69 --------------
 astroquery/vamdc/core.py                                    |   4 +
 astroquery/vamdc/tests/test_vamdc.py                        |  20 ----
 astroquery/vamdc/tests/test_vamdc_remote.py                 |   6 ++
 docs/esa/hubble.rst                                         |  88 +++++++++++++++--
 docs/heasarc/heasarc.rst                                    |  56 +++++++++++
 docs/index.rst                                              |   4 +-
 setup.cfg                                                   |   4 +-
 31 files changed, 1362 insertions(+), 1319 deletions(-)
 create mode 100644 astroquery/esa/hubble/tests/data/cone_search.vot
 create mode 100644 astroquery/esa/hubble/tests/data/cone_search_m31_5.vot
 create mode 100644 astroquery/esa/hubble/tests/data/cone_search_m31_5.vot.test
 delete mode 100644 astroquery/esa/hubble/tests/dummy_handler.py
 create mode 100644 astroquery/esa/hubble/tests/test_esa_hubble_remote.py
 create mode 100644 astroquery/heasarc/tests/test_heasarc_remote_isdc.py
 delete mode 100644 astroquery/lcogt/__init__.py
 delete mode 100644 astroquery/lcogt/core.py
 delete mode 100644 astroquery/lcogt/tests/__init__.py
 delete mode 100644 astroquery/lcogt/tests/data/Box.xml
 delete mode 100644 astroquery/lcogt/tests/data/Cone.xml
 delete mode 100644 astroquery/lcogt/tests/data/Cone_coord.xml
 delete mode 100644 astroquery/lcogt/tests/data/Polygon.xml
 delete mode 100644 astroquery/lcogt/tests/setup_package.py
 delete mode 100644 astroquery/lcogt/tests/test_lcogt.py
 delete mode 100644 astroquery/lcogt/tests/test_lcogt_remote.py
 delete mode 100644 astroquery/vamdc/tests/test_vamdc.py 
```

My `nasa_exoplanet_archive/core.py` doesn't look quite right anymore.

bsipocz commented:
I'll have a gap between meetings and teaching, will give it a go and let you know where to double check before pushing it to this pr

rickynilsson commented:
> I'll have a gap between meetings and teaching, will give it a go and let you know where to double check before pushing it to this pr

Thanks! Sorry for the mess.

bsipocz commented:
No worries, this is a very common issue, and a long-standing feature request towards GH to provide a one-click solution to rebase rather than default to merge for any conflict resolution.

rickynilsson commented:
Yeah, and I wish I didn't even have to rebase for just changing an old commit message. It forces me to rewrite the history for anything coming after that commit. I think something might have gone wrong in the rebase, not sure, ugh.

rickynilsson commented:
As far as I can see, only some minor things in nasa_exoplanet_archive/core.py got messed up (not sure exactly when or how, weirdly the backup before resbase looks the same), and accidentally deleted a "main" in the word "maintained" in CHANGES.rst when I resolved a conflict there. But I'll wait for you to confirm I didn't mess up anything else.

rickynilsson commented:
After the pull, I also have a `__pycache__` directory with `.pyc` files in each directory, which I think is causing problems when I'm running my tests now.

For example: `NasaExoplanetArchive.query_object("HAT-P-11 b", where="nothing")` throws a `Error 904: 42S22 :[Oracle][ODBC][Ora]ORA-00904: "PL_HOSTNAME": invalid identifier` in with `pytest`, but returns the expected `WARNING: InputWarning: Any filters using the 'where' argument are ignored in ''query_object''. Consider using ''query_criteria'' instead. [astroquery.nasa_exoplanet_archive.core]` when running in iPython.

bsipocz commented:
You can always run a `git clean -dfx` to clear all files not part of the repo. Be careful to add ` -e` and path, if you have anything that you keep in the repo uncommitted and want to keep (E.g. I have a notes directory everywhere to store random stuff, and also pycharm project files, so usually just run: `git clean -dfx -e .idea -e notes`)

rickynilsson commented:
Any ideas why the order in which test files are run would affect the results?

Case 1:
```
~/anaconda3/envs/astroquery_test/lib/python3.8/site-packages/astroquery-0.4.3.dev6850-py3.8.egg/astroquery/nasa_exoplanet_archive/tests » pytest test_nasa_exoplanet_archive_remote.py test_nasa_exoplanet_archive.py

...

collected 50 items                                                                                                                                       

test_nasa_exoplanet_archive_remote.py .............            
test_nasa_exoplanet_archive.py ....F................................
```

Case 2:
```
~/anaconda3/envs/astroquery_test/lib/python3.8/site-packages/astroquery-0.4.3.dev6850-py3.8.egg/astroquery/nasa_exoplanet_archive/tests » pytest test_nasa_exoplanet_archive.py test_nasa_exoplanet_archive_remote.py

...

collected 50 items                                                                                                                                       

test_nasa_exoplanet_archive.py .....................................     
test_nasa_exoplanet_archive_remote.py .......F..F.F
```

If I run them separately, both pass.

rickynilsson commented:
@bsipocz, @keflavich, @dfm - All lights are green! Feel free to merge at your earliest convenience. Thanks for all the assistance!

rickynilsson commented:
@bsipocz, @keflavich - just a gentle reminder :)

bsipocz commented:
I've rebased to get rid of the unrelated and duplicated commits, and did some minor cleanup. However, given that the PR was opened from an org account rather than a personal account I cannot force push back to the branch as a maintainer. 
So, while for independent reasons I'll added to that org account, we may also think about how to get access for the other two maintainers (I'm happy to be the responsible maintainer for any IPAC stuff, so it probably won't come up as a blocker)

bsipocz commented on astroquery/nasa_exoplanet_archive/core.py:
```diff
@@ -59,10 +72,40 @@
+# return errors and urge the user to call the 'ps' or 'pscomppars' tables
+OBJECT_TABLES = {"ps": "pl_", "pscomppars": "pl_", "exoplanets": "pl_",
+                 "compositepars": "fpl_", "exomultpars": "mpl_"}
+MAP_TABLEWARNINGS = {"exoplanets": "Planetary Systems (PS)",
```
I think this is a bit confusing in the warning message, but can be adjusted if there are any user complaints.

bsipocz reviewed (approved):
Ready to go once CI is happy.

bsipocz commented:
OK, so the only remaining issue is with the heisen failure of remote tests. 3 of them are failing if they are run after the mock tests, but they all pass when only the remotes are run. 
I suggest to open a separate issue for it, and follow-up in a separate PR. I don't yet see whether it's a bug in the mock tests, or the code, or rather something related to how pytest is run. I kind of rules out it being cache related, as the tests pass even after they failed, if they are run on their own.

bsipocz commented:
Thanks @rickynilsson! 

Would you mind opening that follow-up issue, or prefer me to do it?

rickynilsson commented:
Thank you @bsipocz! I can open the issue about order dependent test fails.

bsipocz commented:
(you can let users know that installing the development version with e.g. `python -m pip install git+https://github.com/astropy/astroquery.git` would pick up this update).
